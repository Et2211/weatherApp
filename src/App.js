import logo from './logo.svg';
import './App.css';
import { connect, useSelector } from 'react-redux';
import { simpleAction } from './actions/simpleActions';
import store from './store'

function App(props) {
  console.log(useSelector(state => state))
  const counter = (useSelector(state => state.simpleReducer.counter))
  console.log(counter)

  const simpleFunction = (event) => {
    console.log(props)
    props.simpleAction();
  }


  return (
    <div className="App">
        <button onClick={simpleFunction}>Test redux action</button>
        <p>{counter}</p>
    </div>
  );
}

const mapStateToProps = state => ({
  ...state
 })

const mapDispatchToProps = dispatch => ({
  simpleAction: () => dispatch(simpleAction())
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
