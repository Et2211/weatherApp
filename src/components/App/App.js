import styles from './App.module.scss';
import { connect, useSelector, useDispatch } from 'react-redux';
import { fetchWeather } from '../../redux/actions/weatherActions';
import store from '../../store'
import { getRequest } from '../../api';
import { useEffect } from 'react';
import Header from '../Header/Header';

function App(props) {

  const dispatch = useDispatch()

  useEffect(()=>{
    refreshLocation()
  }, [])
  
  console.log(useSelector(state => state))

  const counter = (useSelector(state => state.locationsReducer.counter))

  const refreshLocation = (event) => {
    getRequest('/city-data').then(response=>{
      dispatch(fetchWeather(response))
    })
  }


  return (
    <div className={styles.App}>
      <Header />
    </div>
  );
}



export default App;
