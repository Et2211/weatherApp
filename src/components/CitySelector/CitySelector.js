import styles from './CitySelector.module.scss'
import { useState } from 'react';

function CitySelector (props) {
  const [searchInput, setSearchinput] = useState('')

  const searchSubmit = (e) => {
    setSearchinput('')
    props.getWeather(searchInput)
  }
  return (
    <>
      <div className="row justify-content-start">
        <div className="col-6 col-md-3">
          <h2>{props.cityName}</h2>
        </div>
        <div className="col-6 col-md-3">
          <div class={"input-group"}>
            <div class={styles.formOutline +" form-outline"}>
              <input id="search-focus" type="search" class="form-control" value={searchInput} onChange={(e)=>setSearchinput(e.target.value)}/>
              <label class="form-label" for="form1">Search</label>
            </div>
            <button type="button" class="btn btn-primary" onClick={searchSubmit}>
              <i class="fas fa-search"></i>
            </button>
          </div>
          {props.error == 1 && 
            <p className={styles.error}>{props.errorMessage}</p>
          }
            
        </div>
      </div>
    </>
  )
}

export default CitySelector;