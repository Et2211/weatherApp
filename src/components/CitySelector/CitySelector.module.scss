.formOutline{
  border: 1px solid #e2e2e2;
}

.error{
  color: rgb(209, 16, 16);
  font-size: 12px;
}