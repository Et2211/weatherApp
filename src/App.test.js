import { render, screen } from '@testing-library/react'
import App from './components/App/App'
import CitySelector from './components/CitySelector/CitySelector'
import { Provider } from 'react-redux'
import configureStore from './store';
import renderer from 'react-test-renderer'; // ES6
import './setupTests'
const store = configureStore()

jest.mock('./components/CitySelector/CitySelector')
CitySelector.mockImplementation(props => props.user)
jest.mock('./components/DayOverview/DayOverview')

test('1. App renders', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>   
  )
  
  expect(screen.getByText(/Wagonex weather/i)).toBeInTheDocument()
})

test('2. App matches snapshot', () => {
  const appTemplate = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>   
  )
  expect(appTemplate).toMatchSnapshot();

})
