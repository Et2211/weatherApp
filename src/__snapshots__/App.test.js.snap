// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. App matches snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="header"
  >
    Wagonex Weather
  </h1>
  <div
    className="container"
  >
    <div
      className="my-5"
    />
    <div
      className="my-5"
    />
  </div>
</div>
`;
